index.js

```diff
@@ -10,11 +10,26 @@ var onExit = require('signal-exit')
 var path = require('path')
 var activeFiles = {}
 
+// if we run inside of a worker_thread, `process.pid` is not unique
+var id = (function getId () {
```

Comment:
@addaleax sorry to ping you directly, but does this make sense to you?

maybe it makes sense to just generate a uuid or something instead of using `process.pid` at all? I'm not familiar with this project, or problem space

Reply:
@SimenB No problem!

The `threadId` is going to be unique for each Worker inside a process, but for each process it will start at 0, so I think you still want to include the pid in some way?

Reply:
Right, for setups using multiple processes with multiple workers each, something like `${process.pid}-${workerThreads.threadId}` I guess?

Reply:
Right, good call.

I guess I can try to get `threadId`, default to `0` (if either `worker_threads` is not available, or not inside a worker), then add it as a separate `.hash(String(threadId))` while keeping `.hash(String(process.pid))` around?

Reply:
Oh it just goes into hash anyway. I guess no fancy template strings needed then ^^

Reply:
Tried to address this: https://github.com/npm/write-file-atomic/pull/37/commits/7d74261fabb0291aea08b54639fea3fa3fcbf387